libgstd/gstd_parser.c

```diff
@@ -966,3 +980,209 @@ gstd_parser_signal_timeout (GstdSession * session, gchar * action, gchar * args,
 
   return ret;
 }
+
+static GstdReturnCode
+gstd_parser_pipeline_create_ref (GstdSession * session, gchar * action,
+    gchar * args, gchar ** response)
+{
+  gchar **tokens = NULL;
+  gchar *description = NULL;
+  GstdObject *pipeline_list_node = NULL;
+  GstdObject *pipeline_node = NULL;
+  GstdReturnCode ret = GSTD_EOK;
+
+  g_return_val_if_fail (GSTD_IS_SESSION (session), GSTD_NULL_ARGUMENT);
+  g_return_val_if_fail (action, GSTD_NULL_ARGUMENT);
+  g_return_val_if_fail (args, GSTD_NULL_ARGUMENT);
+  g_return_val_if_fail (response, GSTD_NULL_ARGUMENT);
+
+  tokens = g_strsplit (args, " ", 2);
+
+  /* Get the pipeline list node */
+  ret = gstd_get_by_uri (session, "/pipelines", &pipeline_list_node);
+  if (ret) {
+    goto pipeline_list_node_error;
+  }
+
+  GST_OBJECT_LOCK (session);
+
+  /* Look for the pipeline node */
+  pipeline_node =
+      gstd_list_find_child (GSTD_LIST (pipeline_list_node), tokens[0]);
+
+  /* Pipeline doesn't exist */
+  if (!pipeline_node) {
+    ret = gstd_parser_pipeline_create (session, action, args, response);
+    if (ret) {
+      goto create_error;
+    }
+    pipeline_node =
+        gstd_list_find_child (GSTD_LIST (pipeline_list_node), tokens[0]);
+  } else {
+    g_object_get (pipeline_node, "description", &description, NULL);
+    /* Return error code if the descriptions don't match */
+    if (0 != g_strcmp0 (description, tokens[1])) {
+      ret = GSTD_EXISTING_NAME;
+      g_free (description);
+      goto create_error;
+    }
+    g_free (description);
```

Comment:
use a single free, add a new label to jump right before create_error

Reply:
Description is only created inside the `else`. If I create a label to jump from here another label would be needed to skip the free when the program enters the `if`  instead.